src/BugsnagUnity/Native/Android/NativeEvent.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using BugsnagUnity.Payload;
+using UnityEngine;
+
+namespace BugsnagUnity
+{
+    internal class NativeEvent : NativePayloadClassWrapper, IEvent
+    {
+        public NativeEvent(AndroidJavaObject androidJavaObject) : base (androidJavaObject){}
+
+        public string ApiKey { get => GetNativeString("getApiKey"); set => SetNativeString("setApiKey",value); }
+
+        public IAppWithState App => new NativeAppWithState(NativePointer.Call<AndroidJavaObject>("getApp"));
```

Comment:
Does this return a new instance each time the property is accessed, or have I misunderstood C# (again)?

Reply:
It does